pkgs/by-name/va/valkey/package.nix

```diff
@@ -0,0 +1,84 @@
+{ lib, stdenv, fetchFromGitHub, fetchurl, fetchpatch, lua, jemalloc, pkg-config, nixosTests
+, tcl, which, ps, getconf
+, withSystemd ? lib.meta.availableOn stdenv.hostPlatform systemd, systemd
+# dependency ordering is broken at the moment when building with openssl
+, tlsSupport ? !stdenv.hostPlatform.isStatic, openssl
+
+# Using system jemalloc fixes cross-compilation and various setups.
+# However the experimental 'active defragmentation' feature of valkey requires
+# their custom patched version of jemalloc.
+, useSystemJemalloc ? true
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "valkey";
+  version = "7.2.4-rc1";
+
+  src = fetchFromGitHub {
+    owner = "valkey-io";
+    repo = "valkey";
+    rev = finalAttrs.version;
+    hash = "sha256-KH3cCvveH8VyOlc6brAAQ8bleMUJB5uMn6O2mf/HCxc=";
+  };
+
+  patches = [] ++ lib.optionals useSystemJemalloc [./use_system_jemalloc.patch];
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ lua ]
+    ++ lib.optional useSystemJemalloc jemalloc
+    ++ lib.optional withSystemd systemd
+    ++ lib.optionals tlsSupport [ openssl ];
```

Comment:
It would make more sense to just use `lib.optional` here to keep it consistent with the other conditionals.